src/Core/src/Credentials/IniFileProvider.php

```diff
@@ -146,4 +157,68 @@ private function getCredentialsFromRole(array $profilesData, array $profileData,
             Credentials::adjustExpireDate($credentials->getExpiration(), $this->getDateFromResult($result))
         );
     }
+
+    /**
+     * @param array<string, string> $profileData
+     */
+    private function getCredentialsFromLegacySso(array $profileData, string $profile): ?Credentials
+    {
+        if (!isset(
+            $profileData[IniFileLoader::KEY_SSO_START_URL],
+            $profileData[IniFileLoader::KEY_SSO_REGION],
+            $profileData[IniFileLoader::KEY_SSO_ACCOUNT_ID],
+            $profileData[IniFileLoader::KEY_SSO_ROLE_NAME]
+        )) {
+            $this->logger->warning('Profile "{profile}" does not contains require legacy SSO config.', ['profile' => $profile]);
+
+            return null;
+        }
+
+        $ssoCacheFileLoader = new SsoCacheFileLoader($this->logger);
+        $tokenData = $ssoCacheFileLoader->loadSsoCacheFile($profileData[IniFileLoader::KEY_SSO_START_URL]);
```

Comment:
I think a piece of the puzzle is missing:
- This code fetch a token form the cached file
- Then use the client to get a new Role/Crendential

But what is filling the cache?

In the official AwsSdk, there is a code in charge of generating the token and storing it in the cache (see SsoTokenProvider, especially the method `ssoOidcClient->createToken`)

I feel that:
- the cache location is not a AWS standard, but only a file uses by the official PHP-SDK
- this PR is a bridge for the official AWS-SDK, and we still need the SDK to generate the token.

Reply:
There is also a official documentation for SDKs and SSO: https://docs.aws.amazon.com/sdkref/latest/guide/access-sso.html :-)

Reply:
All good with explanation and the linked documentation or are there any open question I might can answer?